### ISSUE TITLE
sf graticules are drawn even if relevant theme elements are set to element_blank()

### ISSUE THREAD

**Issue by karldw** [tidyverse/ggplot2#2525]
`theme_void()` is a great way to clear out all the extra lines in a chart.  My intuition was that this theme would remove to graticules from `geom_sf()` as well, but it doesn't.  Do you think it should?

```r
library(sf) 
library(ggplot2)

nc <- st_read(system.file("gpkg/nc.gpkg", package = "sf")) 

ggplot(nc) + 
  geom_sf(data = nc, aes(fill = AREA)) + 
  theme_void()

# The actual way to remove graticules:
ggplot(nc) + 
  geom_sf(data = nc, aes(fill = AREA)) + 
  coord_sf(datum = NA) + 
  theme_void()
```

Here's a `theme_void` map of North Carolina:
![North Carolina with graticule](https://user-images.githubusercontent.com/12504708/38788362-1e0d215a-40e8-11e8-8d17-3b2ed8959d00.png)

And here's what I expected, with no graticule:
![North Carolina no graticule](https://user-images.githubusercontent.com/12504708/38788354-14030562-40e8-11e8-9866-83542389080f.png)

Ref: https://github.com/tidyverse/ggplot2/issues/2071

**Comment by hadley** [comment]
Duplicate of #2389

**Comment by mikmart** [comment]
@hadley this issue still persists after #2389 and the underlying cause seems to be that `CoordSf$render_bg()` doesn't handle `element_blank()` as the `panel.grid` element properly.

**Comment by clauswilke** [comment]
I can confirm that this problem still exists in the current github master. I'm reopening the issue.

``` r
library(sf) 
#> Linking to GEOS 3.6.1, GDAL 2.1.3, proj.4 4.9.3
library(ggplot2)

nc <- st_read(system.file("gpkg/nc.gpkg", package = "sf"), quiet = TRUE) 

ggplot(nc) + 
  geom_sf(data = nc, aes(fill = AREA)) + 
  theme_void()
```

![](https://i.imgur.com/DAjduHm.png)

Created on 2018-09-24 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).
